tests/page/page-keyboard-layouts.spec.ts

```diff
@@ -0,0 +1,363 @@
+  });
+
+  it(`should fire key events on α`, async ({ page }) => {
+    await page.keyboard.press('α');
+    expect(await page.evaluate('getResult()')).toBe(
+        ['Keydown: α KeyA 65 []',
+          'Keypress: α KeyA 945 945 []',
+          'Keyup: α KeyA 65 []'].join('\n'));
+  });
+
+  it(`should type ε on KeyE`, async ({ page }) => {
+    await page.keyboard.press('KeyE');
+    expect(await page.evaluate('getResult()')).toBe(
+        ['Keydown: ε KeyE 69 []',
+          'Keypress: ε KeyE 949 949 []',
+          'Keyup: ε KeyE 69 []'].join('\n'));
+  });
+
+  it(`should fire key events on Σ`, async ({ page }) => {
+    await page.keyboard.press('Σ');
+    expect(await page.evaluate('getResult()')).toBe(
+        ['Keydown: Σ KeyS 83 []',
+          'Keypress: Σ KeyS 931 931 []',
+          'Keyup: Σ KeyS 83 []'].join('\n'));
+  });
+
+  it(`should type Δ on Shift+KeyD`, async ({ page }) => {
+    await page.keyboard.press('Shift+KeyD');
+    expect(await page.evaluate('getResult()')).toBe(
+        ['Keydown: Shift ShiftLeft 16 [Shift]',
+          'Keydown: Δ KeyD 68 [Shift]',
+          'Keypress: Δ KeyD 916 916 [Shift]',
+          'Keyup: Δ KeyD 68 [Shift]',
+          'Keyup: Shift ShiftLeft 16 []'].join('\n'));
+    await expect(page.locator('textarea')).toHaveValue('Δ');
+  });
+});
+
+it.describe(`portuguese keyboard layout`, () => {
+  it.beforeEach(async ({ page, server, toImpl }) => {
+    await page.keyboard.changeLayout('pt-PT');
+    await page.goto(server.PREFIX + '/input/keyboard.html');
+  });
+
+  it(`should type backslash on Backquote`, async ({ page }) => {
+    await page.keyboard.press('Backquote');
+    expect(await page.evaluate('getResult()')).toBe(
+        ['Keydown: \\ Backquote 220 []',
+          'Keypress: \\ Backquote 92 92 []',
+          'Keyup: \\ Backquote 220 []'].join('\n'));
+  });
+
+  it(`should type ! on Shift+Digit1`, async ({ page }) => {
+    await page.keyboard.press('Shift+Digit1');
+    expect(await page.evaluate('getResult()')).toBe(
+        ['Keydown: Shift ShiftLeft 16 [Shift]',
+          'Keydown: ! Digit1 49 [Shift]',
+          'Keypress: ! Digit1 33 33 [Shift]',
+          'Keyup: ! Digit1 49 [Shift]',
+          'Keyup: Shift ShiftLeft 16 []'].join('\n'));
+  });
+});
+
+it.describe(`us keyboard layout`, () => {
+  it.beforeEach(async ({ page, server, toImpl }) => {
+    await page.keyboard.changeLayout('en-US');
+    await page.goto(server.PREFIX + '/input/keyboard.html');
+  });
+
+  it(`should type backslash on Backslash`, async ({ page }) => {
+    await page.keyboard.press('Backslash');
+    expect(await page.evaluate('getResult()')).toBe(
+        ['Keydown: \\ Backslash 220 []',
+          'Keypress: \\ Backslash 92 92 []',
+          'Keyup: \\ Backslash 220 []'].join('\n'));
+  });
+});
+
+it(`should fallback to us on invalid layout format`, async ({ page, toImpl, server }) => {
+  await page.keyboard.changeLayout('invalid');
```

Comment:
I'd throw instead so users are not confused.

Reply:
I thought it also, I only decided for the fallback instead of throwing an exception because there's no standard layout IDs. But since we're going to explicitly document all supported layouts, that makes sense.

Another thing: I don't particularly like the `keyboard.changeLayout` API, it's a destructive call, and I was thinking that maybe I could just create new keyboard instances per layout. Something like:

```js
const ptKeyboard = page.keyboardFor('pt-PT');
await ptKeyboard.press('Tab');
```

Reply:
I did a branch to test the alternative API proposal, you can find it here: https://github.com/microsoft/playwright/commit/eea892fd6368295d1e1cf178a87d6db10418181f